core/src/cli_commands/transfer_oracle_token.rs

```diff
@@ -0,0 +1,259 @@
+    let op = OraclePool::new().unwrap();
+    if let Some(local_datapoint_box_source) = op.get_local_datapoint_box_source() {
+        let prefix = if ORACLE_CONFIG.on_mainnet {
+            NetworkPrefix::Mainnet
+        } else {
+            NetworkPrefix::Testnet
+        };
+        let rewards_destination =
+            AddressEncoder::new(prefix).parse_address_from_str(&rewards_destination_str)?;
+
+        let change_address_str = get_wallet_status()?
+            .change_address
+            .ok_or(TransferOracleTokenActionError::NoChangeAddressSetInNode)?;
+
+        let change_address =
+            AddressEncoder::new(prefix).parse_address_from_str(&change_address_str)?;
+        let unsigned_tx = build_transfer_oracle_token_tx(
+            local_datapoint_box_source,
+            wallet,
+            rewards_destination,
+            current_block_height()? as u32,
+            change_address,
+        )?;
+
+        println!(
+            "YOU WILL BE TRANSFERRING YOUR ORACLE TOKEN TO {}. TYPE 'YES' TO INITIATE THE TRANSACTION.",
+            rewards_destination_str
+        );
+        let mut input = String::new();
+        std::io::stdin().read_line(&mut input)?;
+        if input == "YES" {
+            let tx_id_str = sign_and_submit_transaction(&unsigned_tx)?;
+            println!(
+                "Transaction made. Check status here: {}",
+                ergo_explorer_transaction_link(tx_id_str, prefix)
+            );
+        } else {
+            println!("Aborting the transaction.")
+        }
+        Ok(())
+    } else {
+        Err(TransferOracleTokenActionError::NoLocalDatapointBox)
+    }
+}
+fn build_transfer_oracle_token_tx(
+    local_datapoint_box_source: &dyn LocalDatapointBoxSource,
+    wallet: &dyn WalletDataSource,
+    oracle_token_destination: Address,
+    height: u32,
+    change_address: Address,
+) -> Result<UnsignedTransaction, TransferOracleTokenActionError> {
+    let in_oracle_box = local_datapoint_box_source.get_local_oracle_datapoint_box()?;
+    let num_reward_tokens = *in_oracle_box.reward_token().amount.as_u64();
+    if num_reward_tokens != 1 {
+        return Err(
+            TransferOracleTokenActionError::IncorrectNumberOfRewardTokensInOracleBox(
+                num_reward_tokens as usize,
+            ),
+        );
+    }
+    if let Address::P2Pk(p) = &oracle_token_destination {
+        // Build the new oracle box
+        let mut builder = ErgoBoxCandidateBuilder::new(
+            in_oracle_box.get_box().value,
+            in_oracle_box.get_box().ergo_tree.clone(),
+            height,
+        );
+        let ec_point = *p.h.clone();
+        builder.set_register_value(R4, ec_point.into());
+        builder.set_register_value(R5, (in_oracle_box.epoch_counter() as i32).into());
+        builder.set_register_value(R6, (in_oracle_box.rate() as i64).into());
+        builder.add_token(in_oracle_box.oracle_token().clone());
+
+        let single_reward_token = Token {
```

Comment:
The command will exit early if the number of existing reward tokens != 1 (line 123 above). I was following what you wrote in the issue: "to avoid accidental reward token transfers let's exit if there is more the one reward token in the input oracle box". So I'm also exiting if there are no reward tokens too.

Reply:
Oh, my bad. I totally missed that check on line 123. `single_reward_token` caught my eye first and I started to search where the rest of the reward tokens go. Do we need `single_reward_token`? I mean if we check that there is only one reward token in `in_oracle_box` it makes sense to just pass it on, i.e. "preserve" the reward token.

Reply:
Oh yes of course, was staring me in the face and I couldn't see it!

Reply:
Thanks a lot, just pushed the fix.